### PR TITLE
Add function to evaluate Git repo name

### PR DESCRIPTION
**Pull request by johnwatson484**
Add function to evaluate Git repo name from Url and return to calling pipeline through `getVariables`

https://eaflood.atlassian.net/browse/PSD-650
https://eaflood.atlassian.net/browse/PSD-651

### CODE REVIEW

File: vars/build.groovy [hunk 1]
```diff
@@ -17,6 +18,11 @@ def getRepoUrl() {
   return sh(returnStdout: true, script: "git config --get remote.origin.url").trim()
 }
 
+// private
+def getRepoName(repoUrl) {
```
Comment (st3v3nhunt) [inline vars/build.groovy:22]: I can't see where `repoUrl` is used, is it necessary?

File: vars/build.groovy [hunk 2]
```diff
@@ -17,6 +18,11 @@ def getRepoUrl() {
...
+  return scm.getUserRemoteConfigs()[0].getUrl().tokenize('/').last().split("\\.")[0]
```
Comment (st3v3nhunt) [inline vars/build.groovy:23]: The scenario might not happen or be very unlikely but if the work stream the repo was for was delimited by periods or there was is another reason for periods to be included in the repo name, the same value would be returned e.g. `ffc.some-service` and `fcc.another-repo` would both return `ffc`. This [comment](https://stackoverflow.com/questions/45684941/how-to-get-repo-name-in-jenkins-pipeline#comment97915730_45690925) suggests `...split("\\.git")[0]` to ensure the full repo name is retrieved.

File: vars/build.groovy (hunk 1 above)
Comment (johnwatson484) [reply vars/build.groovy:22]: It's used when passing the status of the build to github

File: vars/build.groovy (hunk 2 above)
Comment (johnwatson484) [reply vars/build.groovy:23]: That makes sense.  Our naming convention shouldn't allow that to happen anyway but it still seems sensible to add it.

Comment (johnwatson484) [reply vars/build.groovy:23]: Added

**Pull request merged**